### PR TITLE
[Develop] Support for RHEL9 and Rocky9

### PR DESCRIPTION
**Pull request by hanwen-pcluste**
### Description of changes
* See commit descriptions for details

### Tests
* Tests in https://github.com/aws/aws-parallelcluster/blob/develop/tests/integration-tests/configs/new_os.yaml have been run. No failures related to CLI.
* I compared this PR with https://github.com/aws/aws-parallelcluster/pull/5423/files to make sure all changes are included

### References
* Corresponding cookbook PR is working-in-progress https://github.com/aws/aws-parallelcluster-cookbook/pull/2617

### Checklist
- Make sure you are pointing to **the right branch**.
- If you're creating a patch for a branch other than `develop` add the branch name as prefix in the PR title (e\.g\. `[release-3.6]`).
- Check all commits' messages are clear, describing what and why vs how.
- Make sure **to have added unit tests or integration tests** to cover the new/modified code.
- Check if documentation is impacted by this change.

Please review the [guidelines for contributing](../CONTRIBUTING.md) and [Pull Request Instructions](https://github.com/aws/aws-parallelcluster/wiki/Git-Pull-Request-Instructions).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: tests/integration-tests/tests/cloudwatch_logging/test_cloudwatch_logging.py [hunk 1]
```diff
@@ -356,7 +356,15 @@ def _write_dummy_message_to_log(self, log_path, node_type):
...
         # Append the dummy entry to the log
-        cmd = f"sudo tee -a {log_path} < {dummy_log_message_path}"
+        log_file_owner_result = self.remote_command_executor.run_remote_command(
+            f"stat -c '%U' {log_path}", raise_on_error=False
+        )
+        if log_file_owner_result.failed:
+            # If failed, it means the `log_path` does not contain files/directories. Use root as the default owner.
+            log_file_owner = "root"
+        else:
+            log_file_owner = log_file_owner_result.stdout.strip()
```
Comment (dreambeyondorange) [inline tests/integration-tests/tests/cloudwatch_logging/test_cloudwatch_logging.py]: Looks like this is repeated, perhaps this should defined as a function? a referenced here and above?

File: tests/integration-tests/tests/storage/test_fsx_lustre.py [hunk 2]
```diff
@@ -525,6 +525,7 @@ def _create_fsx_lustre_volume_ids(num_existing_fsx_lustre, fsx_factory, import_p
             DeploymentType="PERSISTENT_1",
             PerUnitStorageThroughput=200,
         ),
+        FileSystemTypeVersion="2.12",
```
Comment (dreambeyondorange) [inline tests/integration-tests/tests/storage/test_fsx_lustre.py:528]: I'm not sure we should specify the version type.  Can we not let it use the default?

Comment (hanwen-pcluste) [reply tests/integration-tests/tests/storage/test_fsx_lustre.py:528]: Default from CloudFormation is 2.10. Default from AWS console is 2.12. So we have to specify the version

File: tests/integration-tests/tests/cloudwatch_logging/test_cloudwatch_logging.py (hunk 1 above)
Comment (hanwen-pcluste) [reply tests/integration-tests/tests/cloudwatch_logging/test_cloudwatch_logging.py]: Done

File: tests/integration-tests/remote_command_executor.py [hunk 3]
```diff
@@ -233,7 +233,7 @@ def get_remote_files(self, *args, **kwargs):
     def clear_log_file(self, path: str):
         """Clear a log file in a specific path."""
 
-        self.run_remote_command(f"sudo truncate -s 0 {path}")
+        self.run_remote_command(f"sudo -u {self.get_user_to_operate_on_file(path)} truncate -s 0 {path}")
```
Comment (gmarciani) [inline tests/integration-tests/remote_command_executor.py:236]: Why do we need this change now? Is it a security improvement or what else?

File: tests/integration-tests/tests/common/utils.py [hunk 4]
```diff
@@ -49,6 +49,8 @@
     "rhel8": {"name": "RHEL-8.8*_HVM-202309*", "owners": ["309956199498", "841258680906", "219670896067"]},
     # FIXME: unpin once Lustre client is available for Rocky 8.9
     "rocky8": {"name": "Rocky-8-EC2-Base-8.8*", "owners": ["792107900819"]},  # TODO add china and govcloud accounts
+    "rhel9": {"name": "RHEL-9.3*_HVM-*", "owners": ["309956199498", "841258680906", "219670896067"]},
```
Comment (gmarciani) [inline tests/integration-tests/tests/common/utils.py]: These accounts are the same used in Rhel8 (same between Rocky8 and Rocky9. What about avoiding duplication and define them in constants?

File: tests/integration-tests/tests/update/test_update/test_dynamic_file_systems_update/storage-stack.yaml [hunk 5]
```diff
@@ -86,6 +86,7 @@ Resources:
     Condition: CreateFsx
     Properties:
       FileSystemType: LUSTRE
+      FileSystemTypeVersion: '2.12'
```
Comment (gmarciani) [inline tests/integration-tests/tests/update/test_update/test_dynamic_file_systems_update/storage-stack.yaml]: Why is this change required just now?

File: tests/integration-tests/remote_command_executor.py (hunk 3 above)
Comment (hanwen-pcluste) [reply tests/integration-tests/remote_command_executor.py:236]: According to the commit description:
> In many of the operating systems ParallelCluster supported, `root` user could bypass any files permissions. However, with RHEL9 and Rocky9, `root` user could not bypass files permissions by default. Therefore, this commit improves integration tests to operate on files with the owner users of the files

File: tests/integration-tests/tests/update/test_update/test_dynamic_file_systems_update/storage-stack.yaml (hunk 5 above)
Comment (hanwen-pcluste) [reply tests/integration-tests/tests/update/test_update/test_dynamic_file_systems_update/storage-stack.yaml]: Lustre compatibility: https://docs.aws.amazon.com/fsx/latest/LustreGuide/lustre-client-matrix.html
Basically, RHEL9 uses Lustre client 2.15, which supports Lustre server 2.12, 2.15

File: tests/integration-tests/tests/common/utils.py (hunk 4 above)
Comment (hanwen-pcluste) [reply tests/integration-tests/tests/common/utils.py]: Done

**Review by gmarciani — DISMISSED** [review]

File: tests/integration-tests/conftest.py [hunk 6]
```diff
@@ -1407,14 +1407,14 @@ def odcr_stack(
         AvailabilityZone=availability_zone_1,
         InstanceCount=2,
         InstancePlatform="Linux/UNIX",
-        InstanceType="t3.micro",
+        InstanceType="c5.xlarge",
```
Comment (gmarciani) [inline tests/integration-tests/conftest.py:1410]: Why are we changing the instance types?

**Review by gmarciani — DISMISSED** [review]

File: tests/integration-tests/conftest.py (hunk 6 above)
Comment (hanwen-pcluste) [reply tests/integration-tests/conftest.py:1410]: Because .micro instance types is too small for RHEL9. RHEL 9 requires 1.5GB of memory. t3.micro has 1 GB

See https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/9/html/performing_a_standard_rhel_9_installation/system-requirements-reference_installing-rhel

**Review by gmarciani — APPROVED** [review]
